src/tools/ScenarioMeasurement/Startup/ParserUtility.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+namespace ScenarioMeasurement
+{
+    class ParserUtility
+    {
+
+        public static bool MatchProcessStart(TraceEvent evt, TraceSourceManager source, string processName, IList<int> pids, string commandLine)
+        {
+            return MatchCommandLine(evt, source, commandLine) &&
+                   MatchProcessName(evt, source, processName) &&
+                   MatchProcessID(evt, source, pids);
+        }
+
+        public static bool MatchCommandLine(TraceEvent evt, TraceSourceManager source, string commandLine)
+        {
+            if (source.IsWindows)
+            {
+                if (((string)GetPayloadValue(evt, "CommandLine")).Trim() != commandLine)
+                {
+                    return false;
+                }
+            }
+            // Match the command line as well because pids might be reused during the session
+            return true;
+        }
+
+        public static bool MatchProcessName(TraceEvent evt, TraceSourceManager source, string processName)
+        {
+            if (source.IsWindows)
+            {
+                if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // 15 characters is the maximum length of a process name in Linux kernel event payload
+                if (processName.Length < 15)
+                {
+                    if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return false;
+                    }
+                }
+                else if (!processName.Substring(0, 15).Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should we be alerting the user of the library in some way that they are not getting the behavior they are expecting. Not sure how likely we are to have collisions on process names, but this feels like a place an insidious bug could get in.

Reply:
Yes, it is unlikely we will have another process name that has the same 15 characters but we can avoid the bug by checking the filenames. Do you mean the expected behavior is it should get the full length of the process name?

Reply:
I mean that if my filename is over fifteen characters I might not be aware of the 15 character limit on Linux. You could imagine someone making filenames like ThisIsFifteenCharactersOrMore1, 2, 3, etc... Then that person would be quite confused why there tests are not working the way the expected. I don't think we need to fail or anything, but just writing to the log that we had to truncate the file name seems good, as we are not using the full input that we were given.

Reply:
I see! So far the user's not able to input a filename to our program, and our console output does not print out the event payload either. We are just using the truncated filename to do the matching, and full filename anywhere else. But it's good to keep in mind in case we run into it in the future.